commands/configure_authentication.go

```diff
@@ -18,12 +21,13 @@ type ConfigureAuthentication struct {
 	service configureAuthenticationService
 	logger  logger
 	Options struct {
-		Username             string `long:"username"              short:"u"  env:"OM_USERNAME"  required:"true" description:"admin username"`
-		Password             string `long:"password"              short:"p"  env:"OM_PASSWORD"  required:"true" description:"admin password"`
-		DecryptionPassphrase string `long:"decryption-passphrase" short:"dp"                    required:"true" description:"passphrase used to encrypt the installation"`
-		HTTPProxyURL         string `long:"http-proxy-url"                                                      description:"proxy for outbound HTTP network traffic"`
-		HTTPSProxyURL        string `long:"https-proxy-url"                                                     description:"proxy for outbound HTTPS network traffic"`
-		NoProxy              string `long:"no-proxy"                                                            description:"comma-separated list of hosts that do not go through the proxy"`
+		ConfigFile           string `                             long:"config"                short:"c"                    description:"path to yml file containing authentication configuration"`
+		Username             string `yaml:"username"              long:"username"              short:"u"  env:"OM_USERNAME" description:"admin username"`
```

Comment:
Since we're already diving into the reflection rabbit hole, can we do something fancy to avoid needing to copy the `long` struct tag into the `yaml` tag? Guessing reflection gives you a way to programmatically add struct tags, might be possible to add the `yaml` tag on the fly before passing to yaml parser.

Reply:
the `reflect.tag` only supports read, not write unfortunately. 
An alternative way is to unmarshal the given config file into a `map[string]interface{}`, then inspect what `long` tag the field has and back-fill the corresponding map item into the struct.

It essentially achieves the effect of eliminating `yaml` tag, but means we put a little bit more unmarshal logic into the function instead of in the `yaml.unmarshal`